### ISSUE TITLE
[2020 NeurIPS] CompRess: Self-Supervised Learning by Compressing Representations

### ISSUE THREAD

**Issue by Jasonlee1995** [Jasonlee1995/AI_Papers#73]
Self-supervised learning으로 학습한 teacher 모델을 knowledge distillation을 이용하여 student 모델을 학습시키는 방법

Large memory bank를 위해 MoCo를 이용하여 학습

논문의 핵심 내용과 실험 결과만 간단하게 소개

## 1. Method
<img width="1244" alt="image" src="https://user-images.githubusercontent.com/49643709/171251902-b374f88d-4d40-47d5-ae1b-b1e08ee47986.png">

핵심은 teacher가 이미지들 간에 갖는 관계를 student로 전이하는 것임

과정은 다음과 같음

1. 여러 이미지들에 대하여 teacher, student 모델을 이용하여 embedding vector를 얻음
2. Teacher, student embedding vector를 l2 norm을 이용하여 normalize
3. Teacher와 student의 embedding vector들을 이용하여 이미지들 간의 관계를 구함 (cosine similarity 사용)
4. 이미지들의 관계에 대한 정보를 softmax를 이용하여 확률 분포로 변경 (softmax with temperature)
5. 두 확률 분포가 같아지도록 학습 (teacher model은 frozen, KL divergence를 이용하여 학습)

논문에서 성능을 올리는데 있어 2가지 중요한 점을 꼽음

1. Memory bank : small batch size를 이용하면 성능이 좋지 않아, FIFO 구조의 memory bank를 사용
2. Temperature parameter : small temperature value를 이용 (query와 faraway points가 아닌 close neighborhoods과의 relationship을 볼 수 있도록)

논문에서는 teacher의 memory bank를 이용하는지 아닌지에 따라 2가지 방법을 제시함

1. Ours-2q : teacher, student의 memory bank에는 각 모델들의 representation이 담김
2. Ours-1q : teacher의 memory bank만을 이용 (student는 teacher의 memory bank를 이용하여 similarity 계산)

또한 학습 속도를 빨리하기 위해 teacher의 image에 대한 feature를 저장해놓는 caching을 이용

## 2. Experiments

Experiments setting 중에서 주목할만한 점은 linear probing을 하는데 있어 hyperparameter tuning을 하는데 시간을 줄이기 위해서 feature vector를 l2 norm으로 normalize하고 각 dimension이 zero mean, unit variance가 되게 shift and scale을 함

Experiments 중에서 주목할 점은 `Ours-2q` model은 student embedding 제약이 덜해서 teacher embedding을 완벽히 따라가지 않기에 `Ours-1q` 모델보다 transfer dataset에서 generalize를 잘함
(ImageNet linear probing에서는 `Ours-1q`모델이 성능이 더 좋음)

Ablation 결과는 다음과 같음
1. Small temperature를 사용하는 것이 중요 (논문에서 0.04로 설정)
3. Large memory bank를 사용하는 것이 중요
4. Momentum parameter는 중요하지 않음 (momentum 없애도 잘됨)
5. Caching해도 성능 차이가 얼마 나지 않음 (학습 속도는 훨씬 빠름)

<img width="1228" alt="image" src="https://user-images.githubusercontent.com/49643709/171257813-d711405f-c6ae-4f98-a725-2101b02879db.png">
<img width="1229" alt="image" src="https://user-images.githubusercontent.com/49643709/171259541-b62fcc95-b640-4ae3-948e-0f2e764a6e65.png">
<img width="1223" alt="image" src="https://user-images.githubusercontent.com/49643709/171259602-08eb938c-fe8e-4ed5-82b0-6cb3c1de5622.png">
<img width="1215" alt="image" src="https://user-images.githubusercontent.com/49643709/171259656-7a0a5f7a-becc-4dd9-8809-1537415dd7bf.png">
<img width="1224" alt="image" src="https://user-images.githubusercontent.com/49643709/171259694-4b3cdbea-6b1e-43bb-9989-61ec7c8be9a0.png">
<img width="1222" alt="image" src="https://user-images.githubusercontent.com/49643709/171259729-4d72e12e-20c1-41e8-a668-19a2b52da43e.png">
<img width="1225" alt="image" src="https://user-images.githubusercontent.com/49643709/171259773-77aceea0-f7eb-4fdc-bd26-c5e7042978d9.png">
